### PR TITLE
[TACHYON 603] Make it able to see block distribution of a file in web UI

### PR DESCRIPTION
**Pull request by stormspirit**
Now when we click a file, we can see a Locations column for a file block that lists the workers hosting the data block.

### CODE REVIEW

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 1]
```diff
@@ -219,7 +219,8 @@ private UiFileInfo getUiFileInfo(TachyonFS tachyonClient, int fileId, TachyonURI
         StorageLevelAlias storageLevelAlias =
             StorageDirId.getStorageLevelAlias(blockMeta.getParentDir().getStorageDirId());
         // The block last access time is not available. Use -1 for now.
-        uiFileInfo.addBlock(storageLevelAlias, blockId, blockSize, -1);
+        // The location information is also not available for now.
+        uiFileInfo.addBlock(storageLevelAlias, blockId, blockSize, -1, null);
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: In this case, isn't the worker just showing its own blocks?

File: servers/src/main/java/tachyon/web/UiBlockInfo.java [hunk 2]
```diff
@@ -28,13 +34,27 @@ public UiBlockInfo(BlockInfo blockInfo) {
...
     mLastAccessTimeMs = -1;
+    List<NetAddress> workers = blockInfo.getLocations(new TachyonConf());
+    if (workers != null) {
+      addLocations(workers);
+    }
   }
 
-  public UiBlockInfo(long blockId, long blockLength, long blockLastAccessTimeMs, boolean inMemory) {
+  public UiBlockInfo(long blockId, long blockLength, long blockLastAccessTimeMs,
+                     boolean inMemory, List<NetAddress> locations) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/web/UiBlockInfo.java]: We use four space indentation in cases like this.

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java (hunk 1 above)
Comment (stormspirit) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: Yes, in this case when you click the block info from a worker it only shows id, tier, and size. So I guess I'll just change the comment to be "It's unnecessary to pass a location information here"?

File: servers/src/main/java/tachyon/web/UiBlockInfo.java (hunk 2 above)
Comment (stormspirit) [inline servers/src/main/java/tachyon/web/UiBlockInfo.java]: four more space before boolean?

Comment (calvinjia) [inline servers/src/main/java/tachyon/web/UiBlockInfo.java]: Oh no, I mean just four spaces in total

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java (hunk 1 above)
Comment (calvinjia) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: Sounds good, thanks.

**Comment by uronce-cc** [comment]
+1 LGTM, thanks for the patch!

File: servers/src/main/java/tachyon/web/WebInterfaceBrowseServlet.java [hunk 3]
```diff
@@ -156,6 +158,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       }
       request.setAttribute("currentDirectory", currentFileInfo);
       request.setAttribute("blockSizeByte", currentFileInfo.getBlockSizeBytes());
+      request.setAttribute("workerWebPort", mTachyonConf.getInt(Constants.WORKER_WEB_PORT,
+              Constants.DEFAULT_WORKER_WEB_PORT));
```
Comment (gpang) [inline servers/src/main/java/tachyon/web/WebInterfaceBrowseServlet.java]: I think this should only be indented 4 spaces.

**Comment by gpang** [comment]
Thanks for the PR! I just had a minor comment. Do you have a screenshot showing off this feature?

**Comment by stormspirit** [comment]
screen shot 2015-08-10 at 9 31 31 pm

This is what the detailed information will look like when you click a file. If a block is held by multiple workers, the location column will show a comma-separated list of workers.

**Comment by gpang** [comment]
The screenshot looks great!

**Comment by gpang** [comment]
LGTM

**Pull request merged**